server/src/main/java/org/elasticsearch/http/AbstractHttpServerTransport.java

```diff
@@ -156,17 +169,89 @@ static int resolvePublishPort(Settings settings, List<TransportAddress> boundAdd
+         * underlying exception that caused us to treat the request as bad.
+         */
+        final RestRequest restRequest;
+        {
+            RestRequest innerRestRequest;
+            try {
+                innerRestRequest = RestRequest.request(xContentRegistry, httpRequest, httpChannel);
+            } catch (final RestRequest.ContentTypeHeaderException e) {
+                badRequestCause = getException(badRequestCause, e);
+                innerRestRequest = requestWithoutContentTypeHeader(httpRequest, httpChannel, badRequestCause);
+            } catch (final RestRequest.BadParameterException e) {
+                badRequestCause = getException(badRequestCause, e);
+                innerRestRequest =  RestRequest.requestWithoutParameters(xContentRegistry, httpRequest, httpChannel);
+            }
+            restRequest = innerRestRequest;
+        }
+
+        /*
+         * We now want to create a channel used to send the response on. However, creating this channel can fail if there are invalid
+         * parameter values for any of the filter_path, human, or pretty parameters. We detect these specific failures via an
+         * IllegalArgumentException from the channel constructor and then attempt to create a new channel that bypasses parsing of these
+         * parameter values.
+         */
+        final RestChannel channel;
+        {
+            RestChannel innerChannel;
+            ThreadContext threadContext = threadPool.getThreadContext();
+            try {
+                innerChannel = new DefaultRestChannel(httpChannel, httpRequest, restRequest, bigArrays, handlingSettings, threadContext);
+            } catch (final IllegalArgumentException e) {
+                badRequestCause = getException(badRequestCause, e);
+                final RestRequest innerRequest = RestRequest.requestWithoutParameters(xContentRegistry, httpRequest, httpChannel);
+                innerChannel = new DefaultRestChannel(httpChannel, httpRequest, innerRequest, bigArrays, handlingSettings, threadContext);
+            }
+            channel = innerChannel;
+        }
+
+        dispatchRequest(restRequest, channel, badRequestCause);
+    }
+
+    private RestRequest requestWithoutContentTypeHeader(HttpRequest httpRequest, HttpChannel httpChannel, Exception badRequestCause) {
+        HttpRequest httpRequestWithoutContentType = httpRequest.removeHeader("Content-Type");
+        try {
+            return RestRequest.request(xContentRegistry, httpRequestWithoutContentType, httpChannel);
+        } catch (final RestRequest.BadParameterException e) {
+            badRequestCause.addSuppressed(e);
+            return RestRequest.requestWithoutParameters(xContentRegistry, httpRequestWithoutContentType, httpChannel);
+        }
+    }
+
+    private static Exception getException(Exception firstException, Exception secondException) {
```

Comment:
you can use `org.apache.lucene.util.IOUtils#useOrSuppress` here instead. This is in core so we should be fine